ktor-client/ktor-client-features/ktor-client-retry/common/src/io/ktor/client/fatures/retry/Retry.kt

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.fatures.retry
+
+import io.ktor.client.*
+import io.ktor.client.fatures.retry.Retry.*
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import io.ktor.util.*
+import io.ktor.util.network.*
+import io.ktor.util.pipeline.*
+import io.ktor.utils.io.*
+import io.ktor.utils.io.errors.*
+import kotlinx.coroutines.*
+import kotlin.time.*
+
+/**
+ * This feature provides ability to recover from network errors
+ * identified as [IOException] and [HttpRequestTimeoutException] and [UnresolvedAddressException].
+ * It also may recover from request validation that usually produce [ResponseException]
+ * on a non-successful response status code.
+ *
+ * The number of retries is limited by [retries] property.
+ * After every failed attempt it does a fixed time delay of [retryIntervalInSeconds].
+ *
+ * When all attempts exceeded, [RequestRetriesExceededException] exception is thrown
+ * having a list of all errors in the suppressed list and/or a cause.
+ *
+ * @property retries number of retries to recover
+ * @property retryIntervalInSeconds between attempts
+ */
+public class Retry private constructor(
```

Comment:
A couple of design questions: 
* Should the list of allowed exceptions be configurable?
* Can we enable/disable retries per request? Can be a flag in request builder that saves some flag in the request attributes

Reply:
Don't know how to configure it as people usually need subtypes that is not possible with KClass while lambdas have problems with freezes. Any ideas?
Regarding the flag: any ideas, what use-cases? How it should look like? Only inside of the request builder block, right?